spring-cloud-stream/src/main/java/org/springframework/cloud/stream/annotation/StreamListener.java

```diff
@@ -33,12 +33,17 @@
  * the method is invoked and how their return results are processed. This annotation
  * can be applied for two separate classes of methods.
  *
+ * <h3>Declarative mode</h3>
+ *
+ * Method is considered as declarative if at least one of the method parameters is annotated with {@link Input}
```

Comment:
See my other comment about annotating parameters: I don't think we should accept cases when there is a mix of annotated and not annotated parameters.